### ISSUE TITLE
Segmentation fault when reading an instance from a file

### ISSUE THREAD

**Issue by SelinBayramoglu** [ds4dm/ecole#325]
## Describe the bug
I can use the method env.reset only when the instances are generated from an instance generator. When I use the path of an existing lp as an argument to env.reset, I get a segmentation fault.

## Setting
- OS: Linux
- Compiler:
- Python version: Python 3.9.7
- Ecole version: Ecole 0.8.0

## To Reproduce
```python
import ecole

env = ecole.environment.Branching()
instance = "test.lp"
observation, action_set, reward_offset, done, info = env.reset(instance)
```

## Expected behavior
I was expecting the program to work without errors, since there are examples in this repository where env.reset is called on the path to an instance file, for example env.reset("myinstance.lp").

Any help will be much appreciated!

**Comment by AntoinePrv** [comment]
Hello @SelinBayramoglu,

Unfortunately I cannot reproduce your error, especially without knowing what `test.lp` is.
I would suggest you try to solve `test.lp` with SCIP/PySCIPOpt alone.
- If you have an error, then the problem comes from there.
- If you do not have an error, I think the most likely is that you may have a issue with your installation, as the amount of Ecole code that is involved here is really small and definitely tested.

**Comment by tegg89** [comment]
I also have the same error. I could read the instance problem and optimize it with SCIP and PySCIPOpt. However, when I reset the Ecole environment with the same instance, the system corrupted due to the segmentation fault error.
If I run it with Ecole installed via `conda`, I can run the instance without any error. However, Ecole installed via `pip` causes the segmentation fault error. 

# Setting
- OS: Ubuntu 18.04
- Python version: Python 3.7 (also tried ver. 3.9 and 3.10)
- Ecole version: Ecole 0.8.x for `pip` installation (Ecole 0.7 for `conda` installation)

**Comment by SelinBayramoglu** [comment]
I had this problem with SCIP 8.0 and could fix it by switching to an older version of SCIP. My current working versions are
SCIP 7.0.3 and Ecole 0.7.3, both installed from the conda-forge channel.

**Comment by AntoinePrv** [comment]
> I also have the same error. I could read an instance problem and optimize it with SCIP and PySCIPOpt. However, when I reset the Ecole environment with the same instance, the system corrupted due to the segmentation fault error.

Is this one specific file? Or any file you try with?

> If I run it with Ecole installed via conda, I can run the instance without any error. However, Ecole installed via pip causes the segmentation fault error.

Did you try calling other SCIP functions from Ecole (*e.g.* `ecole.scip.Model.prob_basic()`)? If none of them work with your pip install, it could be an installation problem rather than a problem in the library (Ecole pip installations are hard to get right because SCIP is not part of the environment).

**Comment by amf272** [comment]
Hi, I am running into this segfault as well. I am getting the segfault with this instance:
[instance_0.cip.zip](https://github.com/ds4dm/ecole/files/9749979/instance_0.cip.zip)
Is there any ability to run in debug mode to give a better description of the error?
If it helps I also ran this with a .lp file from a separate instance which failed as well.
On my end I am able to run your snippet without error `ecole.scip.Model.prob_basic()`

I installed ecole from source with scip 8.0.2 and ecole version 0.8.1. Also, on my end it seemed to work in an earlier version of ecole but I installed the newer version as I am starting a new project.

On my end the command that doesn't work is
`ecole.scip.Model.from_file(instance)` rather than the above code (but the above code doesn't work as well)

**Comment by SelinBayramoglu** [comment]
I also solved that problem with using an earlier version of Ecole (0.7.3).

**Comment by tegg89** [comment]
I tried multiple problem files, and the program interrupted with the error message: `Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)`. 
When I use the conventional problem with the implemented generator, i.e., set cover, the environment successfully loads the problem.
I found that the Ecole CAN load the customized problem file in the terminal. However, the environment failed when using in PyCharm.
